TLM/TLM/Util/AutoTimedTrafficLights.cs

```diff
@@ -0,0 +1,410 @@
+        /// The directions toward which the traffic light is green
+        /// </summary>
+        private enum GreenDir {
+            AllRed,
+            AllGreen,
+            RightOnly
+        }
+
+        public enum ErrorResult {
+            Success = 0,
+            NoJunction,
+            NoNeed,
+            TTLExists,
+            Other,
+        }
+
+        /// <summary>
+        /// creats a sorted list of segmetns connected to nodeId.
+        /// roads without outgoing lanes are excluded as they do not need traffic lights
+        /// </summary>
+        /// <param name="nodeId">the junction</param>
+        /// <returns>a list of segments aranged in counter clockwise direction.</returns>
+        private static List<ushort> CWSegments(ushort nodeId) {
+            List<ushort> segList = new List<ushort>();
+            netService.IterateNodeSegments(
+                nodeId,
+                ClockDirection.Clockwise,
+                (ushort segId, ref NetSegment _) => {
+                    if (CountOutgoingLanes(segId, nodeId) > 0) {
+                        segList.Add(segId);
+                    }
+                    return true;
+                });
+;
+            return segList;
+        }
+
+        /// <summary>
+        /// Adds an empty timed traffic light if it does not already exists.
+        /// additionally allocates dedicated turning lanes if possible.
+        /// </summary>
+        /// <param name="nodeId">the junction for which we want a traffic light</param>
+        /// <returns>true if sucessful</returns>
+        public static bool Add(ushort nodeId) {
+            List<ushort> nodeGroup = new List<ushort>(1);
+            nodeGroup.Add(nodeId);
+            return tlsMan.SetUpTimedTrafficLight(nodeId, nodeGroup);
+        }
+
+        /// <summary>
+        /// Creates and configures default traffic the input junction
+        /// </summary>
+        /// <param name="nodeId">input junction</param>
+        /// <returns>true if successful</returns>
+        public static ErrorResult Setup(ushort nodeId) {
+            if(tlsMan.HasTimedSimulation(nodeId)) {
+                return ErrorResult.TTLExists;
+            }
+
+            var segList = CWSegments(nodeId);
+            int n = segList.Count;
+
+            if (n < 3) {
+                return ErrorResult.NoNeed;
+            }
+
+            if (!Add(nodeId)) {
+                return ErrorResult.Other;
+            }
+
+            if (SeparateLanes) {
+                LaneArrowManager.SeparateTurningLanes.SeparateNode(nodeId, out _);
+            }
+
+            //Is it special case:
+            {
+                var segList2Way = TwoWayRoads(segList, out int n2);
+                //Debug.Log($"n2 == {n2}");
+                if (n2 < 2) {
+                    return ErrorResult.NoNeed;
+                }
+                bool b = HasIncommingOneWaySegment(nodeId);
+                //Debug.Log($"HasIncommingOneWaySegment == {b}");
```

Comment:
Please look through the diff and remove Debug.Log's from the final pull request.